doc/syntax.md

```diff
@@ -0,0 +1,476 @@
+# NESTML language documentation
+
+NESTML is a domain specific language for the specification of models for the neuronal simulator [NEST](http://www.nest-simulator.org). It has a concise syntax based on that of Python which avoids clutter in the form of semicolons, curly braces or tags as known from other programming and description languages. Instead it concentrates on domain concepts that help to efficiently write neuron models and their equations.
+
+NESTML files are expected to have the filename extension `.nestml`. Each file may contain one or more neuron models. This means that there is no forced direct correspondence between model and file name. Models that shall be compiled into one extension module for NEST have to reside in the same directory. The name of the directory will be used as the name of the corresponding module.
+
+In order to give users complete freedom in implementing neuron model dynamics, NESTML has a full procedural programming language built in. This programming language is used to define the model's time update and functions.
+
+### Data types and physical units
+
+Data types define types of variables as well as parameters and return values of functions. NESTML provides the following primitive types and physical data types:
+
+#### Primitive data types
+* `real` corresponds to the `double` data type in C++. Example literals are: `42.0`, `-0.42`, `.44`
+* `integer` corresponds to the `long` data type in C++. Example literals are: `42`, `-7`
+* `boolean` corresponds to the `bool` data type in C++. Its only literals are `true` and `false`
+* `string` corresponds to the `std::string` data type in C++. Example literals are: `"Bob"`, `""`, `"Hello World!"`
+* `void` corresponds to the `void` data type in C++. No literals are possible and this can only be used in the declaration of a function without return value.
+
```

Comment:
@DimitriPlotnikov: do we really need the `void` data type? If we can establish that function return types can always be deducted from the type of the returned expression, I think we can also drop this. I think that `void` is a very low-level concept that should not be present in a high-level language as NESTML.